ui/src/components/TestRunner/index.tsx

```diff
@@ -84,38 +84,75 @@ export const TestRunner = () => {
     setFormData(defaultFormData);
   }
 
+  function groupBySep(testRuns: TestCase[]): GroupedTestCases {
+    const groupedTestRuns = [];
+    let currentSep;
+    for (const testRun of testRuns) {
+      if (Number(testRun.test.sep) !== currentSep) {
+        currentSep = Number(testRun.test.sep);
+        groupedTestRuns.push({
+          progress: { running: 0, total: 0 },
+          sep: currentSep,
+          tests: [] as TestCase[],
+        });
+      }
+      const sepGroup = groupedTestRuns[groupedTestRuns.length - 1];
+      sepGroup.progress.total++;
+      sepGroup.tests.push(testRun);
+    }
+    return groupedTestRuns;
+  }
+
   // add/remove websocket listener for getTests on component mount/dismount
   useEffect(() => {
     socket.on("getTests", (tests) => {
       const testRunOrderMap: Record<string, number> = {};
       const testRuns = parseTests(tests);
-      let i = 0;
-      for (const testRun of testRuns)
-        testRunOrderMap[getTestRunId(testRun.test)] = i++;
+      let testIndex = 0;
+      let currentSep = "";
+      for (const testRun of testRuns) {
+        const testRunSep = testRun.test.sep;
+        if (testRunSep !== currentSep) {
+          currentSep = testRunSep;
+          testIndex = 0;
+        }
+        testRunOrderMap[getTestRunId(testRun.test)] = testIndex++;
+      }
+
       setTestRunOrderMap(testRunOrderMap);
-      setTestRunArray(testRuns);
+      setTestRunArray(groupBySep(testRuns));
```

Comment:
we don't actually use the "ungrouped" test runs, so let's not store it. We'll only store tests runs after they're grouped by sep